pkg/ocmagenthandler/ocmagenthandler_networkpolicy.go

```diff
@@ -131,3 +133,81 @@ func (o *ocmAgentHandler) ensureNetworkPolicyDeleted(ocmAgent ocmagentv1alpha1.O
 	}
 	return nil
 }
+
+func buildNetworkPolicyForMUO(ocmAgent ocmagentv1alpha1.OcmAgent) netv1.NetworkPolicy {
+	namespacedName := oah.BuildNamespacedName(ocmAgent.Name + "-muo-communication")
+
+	np := netv1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      namespacedName.Name,
+			Namespace: namespacedName.Namespace,
+		},
+		Spec: netv1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{
+				MatchLabels: map[string]string{"app": ocmAgent.Name},
+			},
+			Ingress: []netv1.NetworkPolicyIngressRule{
+				{
+					From: []netv1.NetworkPolicyPeer{
+						{
+							NamespaceSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{"kubernetes.io/metadata.name": "openshift-managed-upgrade-operator"},
+							},
+						},
+					},
+				},
+			},
+			PolicyTypes: []netv1.PolicyType{
+				netv1.PolicyTypeIngress,
+			},
+		},
+	}
+
+	return np
+}
+
+func (o *ocmAgentHandler) ensureNetworkPolicyForMUO(ocmAgent ocmagentv1alpha1.OcmAgent) error {
+	namespacedName := oah.BuildNamespacedName(ocmAgent.Name + "-muo-communication")
+	foundResource := &netv1.NetworkPolicy{}
+
+	// Check if the resource already exist
+	o.Log.Info("ensuring MUO networkpolicy exists", "resource", namespacedName.String())
+	if err := o.Client.Get(o.Ctx, namespacedName, foundResource); err != nil {
+		if k8serrors.IsNotFound(err) {
+			// If Network policy doesn't exist, create it.
+			o.Log.Info("MUO NetworkPolicy does not exist; will be created.")
+			resource := buildNetworkPolicyForMUO(ocmAgent)
+			if err := controllerutil.SetControllerReference(&ocmAgent, &resource, o.Scheme); err != nil {
+				return err
+			}
+			return o.Client.Create(o.Ctx, &resource)
+		}
+		return err
+	} else {
+		// If Network policy exists, check if it needs updating.
+		resource := buildNetworkPolicyForMUO(ocmAgent)
+		if !reflect.DeepEqual(foundResource.Spec, resource.Spec) {
+			o.Log.Info("MUO NetworkPolicy exists but is outdated. Updating.")
+			foundResource.Spec = *resource.Spec.DeepCopy()
+			return o.Client.Update(context.TODO(), foundResource)
+		}
+	}
+	return nil
+}
+
+func (o *ocmAgentHandler) ensureNetworkPolicyDeletedForMUO(ocmAgent ocmagentv1alpha1.OcmAgent) error {
+	namespacedName := oah.BuildNamespacedName(ocmAgent.Name + "-muo-communication")
```

Comment:
For the networkpolicy naming, it would be better to reflect the real use case explicitly, like `allow-muo-communication`, `deny-muo-communication` etc.

BTW, I am wondering if we need a explicit `deny-all` here to allow the only selected allows. (Out of the scope of this card)

Reply:
> For the networkpolicy naming, it would be better to reflect the real use case explicitly, like `allow-muo-communication`, `deny-muo-communication` etc.

Yeah, it makes sense, will update with a more explicit policy name. Thanks